extensions/amp-mega-menu/validator-amp-mega-menu.protoascii

```diff
@@ -0,0 +1,188 @@
+#
+# Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+tags: {  # amp-mega-menu
+  html_format: AMP
+  tag_name: "SCRIPT"
+  extension_spec: {
+    name: "amp-mega-menu"
+    version: "0.1"
+    version: "latest"
+  }
+  attr_lists: "common-extension-attrs"
+}
+tags: {  # <amp-mega-menu>
+  html_format: AMP
+  tag_name: "AMP-MEGA-MENU"
+  requires_extension: "amp-mega-menu"
+  requires: "amp-mega-menu nav"
```

Comment:
`requires`/`satisfies` may not be doing exactly what you expect. They are whole-document booleans. e.g. I think this will be valid:

```
<amp-mega-menu layout=fixed-height>
  <amp-list></amp-list> <!-- meets the `child_tag_name_oneof` requirement but not the `requires` -->
</amp-mega-menu>
<amp-mega-menu layout=fixed-height>
  <nav><ol>...valid li...</ol></nav> <!-- `satisfies: "amp-mega-menu"` for both mega-menus -->
</amp-mega-menu>
```

If I'm wrong about that, add a test to prove me wrong. :)

But is this necessary? Maybe the mandatory_num_child_tags+child_tag_name_oneof is sufficient?

Reply:
Ah you're right. It'd be simple if we didn't need to support the second case in the comment above. Rather than using requires/satisfies, perhaps I can still use reference points to cover both cases.

Reply:
@twifkak updated the validation rules and added some more tests. I'm now using reference points to validate the two cases mentioned before, and should no longer need to change `validator-main.protoascii`. The only situation not covered by the new rules is nested navs, which I can add as a dev warning instead.